pkg/controller/clusterconnection/clusterconnection_controller.go

```diff
@@ -239,20 +261,19 @@ func (r *ReconcileConnection) Reconcile(ctx context.Context, request reconcile.R
 	for _, secretName := range []string{render.PacketCaptureCertSecret, render.PrometheusTLSSecretName} {
 		secret, err := certificateManager.GetCertificate(r.Client, secretName, common.OperatorNamespace())
 		if err != nil {
-			reqLogger.Error(err, fmt.Sprintf("failed to retrieve %s", secretName))
-			r.status.SetDegraded(fmt.Sprintf("Failed to retrieve %s", secretName), err.Error())
+			status.SetDegraded(r.status, operatorv1.ResourceReadError, fmt.Sprintf("Failed to retrieve %s", secretName), err, reqLogger)
 			return reconcile.Result{}, err
 		} else if secret == nil {
 			reqLogger.Info(fmt.Sprintf("Waiting for secret '%s' to become available", secretName))
-			r.status.SetDegraded(fmt.Sprintf("Waiting for secret '%s' to become available", secretName), "")
+			r.status.SetDegraded(string(operatorv1.ResourceNotReady), fmt.Sprintf("Waiting for secret '%s' to become available", secretName))
```

Comment:
I think we can just use status.setDegraded(), rather than r.status.SetDegraded(). This happens a few times in this file.